### ISSUE TITLE
Cannot upgrade GPaste on Fedora 24

### ISSUE THREAD

**Issue by dcharlespyle** [Keruspe/GPaste#161]
Get the following when I try:

./autogen.sh
src/vapi.mk:32: error: ENABLE_VAPIGEN does not appear in AM_CONDITIONAL
Makefile.am:66:   'src/vapi.mk' included from here
autoreconf: automake failed with exit status: 1

./configure --sysconfdir=/etc

***
checking for APPSTREAM... yes
checking X11/extensions/XInput2.h usability... no
checking X11/extensions/XInput2.h presence... yes
configure: WARNING: X11/extensions/XInput2.h: present but cannot be compiled
configure: WARNING: X11/extensions/XInput2.h:     check for missing prerequisite headers?
configure: WARNING: X11/extensions/XInput2.h: see the Autoconf documentation
configure: WARNING: X11/extensions/XInput2.h:     section "Present But Cannot Be Compiled"
configure: WARNING: X11/extensions/XInput2.h: proceeding with the compiler's result
configure: WARNING:     ## ---------------------------------------- ##
configure: WARNING:     ## Report this to Marc-Antoine@Perennou.com ##
configure: WARNING:     ## ---------------------------------------- ##
checking for X11/extensions/XInput2.h... no
checking for pkg-config... (cached) /usr/bin/pkg-config
checking pkg-config is at least version 0.16... yes
checking for appstream-util... /usr/bin/appstream-util
checking for gobject-introspection... yes
./configure: line 17294: syntax error near unexpected token `vala_required'
./configure: line 17294: `VAPIGEN_CHECK(vala_required)'

make
make: *** No targets specified and no makefile found.  Stop.

Checking for appstream gives:

sudo dnf install appstream
Fedora 24 - x86_64 - Updates                    2.6 MB/s | 4.5 MB     00:01    
Fedora 24 - x86_64 - Test Updates               268 kB/s | 6.5 MB     00:24    
Last metadata expiration check: 0:00:26 ago on Sat Jun 18 22:18:13 2016.
Package appstream-0.9.6-1.fc24.x86_64 is already installed, skipping.
Dependencies resolved.
Nothing to do.
Complete!

sudo dnf install appstream-glib
Last metadata expiration check: 0:00:52 ago on Sat Jun 18 22:18:13 2016.
No package appstream-glib available.
Error: Unable to find a match.

`rpm -qa *appstream*`
libappstream-glib-devel-0.5.14-1.fc24.x86_64
appstream-qt-0.9.6-1.fc24.x86_64
appstream-data-24-6.fc24.noarch
appstream-devel-0.9.6-1.fc24.x86_64
appstream-0.9.6-1.fc24.x86_64
libappstream-glib-0.5.14-1.fc24.x86_64
appstream-qt-devel-0.9.6-1.fc24.x86_64

`rpm -qa *vala*`
vala-tools-0.32.0-1.fc24.x86_64
vala-compat-0.18.1-6.fc24.x86_64
vala-devel-0.32.0-1.fc24.x86_64
valabind-0.9.2-4.fc24.x86_64
vala-0.32.0-1.fc24.x86_64

Reinstalling vala packages and installing some additional devel packages gets me further, but still get the following at the end:

```
  CC       src/libgpaste/applet/lib_libgpaste_la-gpaste-applet.lo
In file included from /usr/include/glib-2.0/glib/deprecated/gthread.h:127:0,
                 from /usr/include/glib-2.0/glib.h:107,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from ./src/libgpaste/core/gpaste-item-enums.h:14,
                 from ./src/libgpaste/client/gpaste-client.h:14,
                 from ./src/libgpaste/applet/gpaste-applet-icon.h:14,
                 from ./src/libgpaste/applet/gpaste-applet.h:14,
                 from src/libgpaste/applet/gpaste-applet.c:7:
/usr/include/sys/types.h:70:17: error: two or more data types in declaration specifiers
 typedef __mode_t mode_t;
                 ^~~
cc1: warning: unrecognized command line option ‘-Wno-unknown-warning-option’
Makefile:2050: recipe for target 'src/libgpaste/applet/lib_libgpaste_la-gpaste-applet.lo' failed
make[2]: *** [src/libgpaste/applet/lib_libgpaste_la-gpaste-applet.lo] Error 1
Makefile:2968: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
Makefile:1354: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by Keruspe** [comment]
Any reason why you're not bulding from a release tarball?

**Comment by dcharlespyle** [comment]
I did.  It didn't work, either.  So, I tried to download using git.  Multiple attempts, using differing methods, all ended in multiple failures on my Fedora 24 system.  Nothing I tried worked.

**Comment by Keruspe** [comment]
Can you retry without running autogen.sh, directly running the configure script and then make?

**Comment by dcharlespyle** [comment]
The configure script gave me the following at the end:

checking X11/extensions/XInput2.h usability... no
checking X11/extensions/XInput2.h presence... yes
configure: WARNING: X11/extensions/XInput2.h: present but cannot be compiled
configure: WARNING: X11/extensions/XInput2.h:     check for missing prerequisite headers?
configure: WARNING: X11/extensions/XInput2.h: see the Autoconf documentation
configure: WARNING: X11/extensions/XInput2.h:     section "Present But Cannot Be Compiled"
configure: WARNING: X11/extensions/XInput2.h: proceeding with the compiler's result
configure: WARNING:     ## ---------------------------------------- ##
configure: WARNING:     ## Report this to Marc-Antoine@Perennou.com ##
configure: WARNING:     ## ---------------------------------------- ##
checking for X11/extensions/XInput2.h... no
checking for pkg-config... (cached) /usr/bin/pkg-config
checking pkg-config is at least version 0.16... yes
checking for appstream-util... /usr/bin/appstream-util
checking for gobject-introspection... yes
checking for pkg-config... (cached) /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for vapigen... yes
checking for DBUS... yes
checking for JS_EXTENSION... yes
checking for GDK_X11... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating po/Makefile.in
config.status: creating config.h
config.status: executing libtool commands
config.status: executing depfiles commands
config.status: executing po/stamp-it commands

    GPaste 3.20.3
    ===========

    prefix:                 /usr
    exec_prefix:            ${prefix}
    sysconfdir:             /etc
    libdir:                 ${exec_prefix}/lib64
    libexecdir:             ${exec_prefix}/libexec
    includedir:             ${prefix}/include
    datarootdir:            ${prefix}/share
    datadir:                ${datarootdir}

    compiler:               gcc
    cflags:                 -g -O2 -pedantic -Wall -Wextra -Wvla -Wundef -Wlogical-op -Wsign-compare -Wformat-nonliteral -Wold-style-definition -Wpointer-arith -Winit-self -Wfloat-equal -Wmissing-prototypes -Wstrict-prototypes -Wredundant-decls -Wmissing-noreturn -Wendif-labels -Wcast-align -Wp,-D_FORTIFY_SOURCE=2 -fno-common -fdiagnostics-show-option -fno-strict-aliasing -fvisibility=hidden -fdata-sections
    ldflags:                 -Wl,--as-needed -Wl,--gc-sections

    vapi generator:         /usr/bin/vapigen-0.32

    GObject Introspection:  yes
    Vapi:                   yes

    Status icon:            yes
    App indicator:          no
    Gnome-Shell extension:  yes

    X keybinder support:    yes

    Control Center dir:     /usr/share/gnome-control-center/keybindings
    DBus user services dir: /usr/share/dbus-1/services
    systemd user units dir: /usr/lib/systemd/user

    Bash completion:        yes
    Bash completion dir:    ${datadir}/bash-completion/completions

    Zsh completion:         yes
    Zsh completion dir:     ${datadir}/zsh/site-functions

I got the following from make after that at the end:

Makefile:2041: recipe for target 'src/libgpaste/applet/lib_libgpaste_la-gpaste-applet.lo' failed
make[2]: *** [src/libgpaste/applet/lib_libgpaste_la-gpaste-applet.lo] Error 1
Makefile:2959: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
Makefile:1345: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by Keruspe** [comment]
Does it build with --disable-applet?

**Comment by dcharlespyle** [comment]
Nope.  Fails with --disable-applet as well.

**Comment by Keruspe** [comment]
Can you paste the full build log?

**Comment by dcharlespyle** [comment]
Can't post it.  Too long.  But, I'll attach a text file containing it.

[gpaste build errors.txt](https://github.com/Keruspe/GPaste/files/323777/gpaste.build.errors.txt)

**Comment by Keruspe** [comment]
The only error I see is

/usr/lib/gcc/x86_64-redhat-linux/6.1.1/include/stddef.h:216:22: error: duplicate ‘unsigned’

Which is a file provided by gcc 6.1.1 which is the current development version of gcc.

Please retry with a stable release of gcc like 5.4.0 or 6.1.0

**Comment by dcharlespyle** [comment]
I'm using Fedora 24.  There is no lower version of gcc available for that at the moment in the repositories.  

DNF gives me:

`sudo dnf downgrade gcc

Last metadata expiration check: 2:07:44 ago on Mon Jun 20 08:30:03 2016.
Package gcc of lowest version already installed, cannot downgrade it.
Error: Nothing to do.`

Looks like I'll have to wait for GA to see if this improves.

**Comment by Keruspe** [comment]
Are you actually able to compile any C project? And if you do, any gnome project?

**Comment by dcharlespyle** [comment]
Yes.  And, I just used an older tarball of GPaste, version 3.20.1.  I was able to compile that and install it as well.  I was using version 3.18.  Going to try 3.20.2 in a just a moment and see what happens.

**Comment by dcharlespyle** [comment]
I was able to build and install version 3.20.2 of GPaste just now.  It is working.

**Comment by dcharlespyle** [comment]
After doing two prior upgrades, I was able to make and install 3.20.3.  I think I may know the cause.  I saw an odd error about one of the directories in ~/.ccache being denied access during the building of 3.20.3.  So, I deleted the entire ~/.ccache directory before attempting the build again and it worked.  Not sure if that was the cause of the problem or whether I needed to do two prior upgrades first.  It's a mystery to me at the moment.  But, it seems to be working now.  Thanks.
